### DISCUSSION TITLE
What's the benefit of using eventloop?

### DISCUSSION THREAD

**Discussion by ghost** [gbeced/basana#13] [General]
Are we expecting a lot of IO operations? Even so, making use of a threadpool for IO shall suffice.
Interested to know the motive behind going with eventloop.

**Comment by ghost** [comment]
After a read through the codebase, I find the async use is to asynchronously generate events on the produce and also consume on the consumer side.
This comes handy for live trading.

**Comment by gbeced** [comment]
Being an event driven framework.. I expect most, if not all, of events (trades, bars, order updates, etc) to come from some sort of external source either via polling some service, or through a websocket, or a database, etc.
It is for that reason is that I consider this framework to be mostly IO bound, and I think that asyncio is a lighter and more expressive option, at least in this particular context.
Having said that.. you're free to use threads, multi-processes, or whatever you think its best.. but note that threads in Python is not going to buy you that much, in terms of parallelism, due to the GIL.
